### ISSUE TITLE
Bad typings for (static) Model.increment and Model.decrement

### ISSUE THREAD

**Issue by cyrilchapon** [sequelize/sequelize#14957]
## Issue Creation Checklist

- [X ] I understand that my issue will be automatically closed if I don't fill in the requested information
- [X ] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

## Bug Description

When using sequelize with Typescript, `Model.increment` and `Model.decrement` (static ones) are wrongly typed.

The [doc here](https://sequelize.org/api/v6/class/src/model.js~model#static-method-increment) states that it would return `Promise<[updatedInstances, count]>` whereas the typing state it returns `Promise<updatedInstance>` (a single one).

<img width="370" alt="Capture d’écran 2022-09-05 à 16 43 58" src="https://user-images.githubusercontent.com/10728426/188474468-00881095-5b0d-478e-b94a-759f4fc3bb4e.png">

The doc is right, I checked :

```bash
[ [ [ [Object] ], 1 ] ]
```

### Reproducible Example

Here is the link to the SSCCE for this issue: https://github.com/sequelize/sequelize-sscce/pull/234

### What do you expect to happen?

I'd expect TS typings to show `Promise<[Model[], number]>`

### What is actually happening?

They show `Promise<Model>`

### Environment

- Sequelize version:
```
yarn list v1.22.19
├─ sequelize-cli@6.4.1
├─ sequelize-pool@7.1.0
└─ sequelize@6.21.4
```
- Node.js version:
```
v18.8.0
```
- If TypeScript related: TypeScript version:
```
typescript@4.8.2
```
- Database & Version:
```
postgres:11.2-alpine
```
- Connector library & Version:
```
pg@8.8.0
```
## Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time and I know how to start.
- [ ] Yes, I have the time but I will need guidance.
- [ ] No, I don't have the time, but my company or I are [supporting Sequelize through donations on OpenCollective](https://opencollective.com/sequelize).
- [X] No, I don't have the time, and I understand that I will need to wait until someone from the community or maintainers is interested in resolving my issue.

---

*Indicate your interest in the resolution of this issue by adding the 👍 reaction. Comments such as "+1" will be removed.*

**Comment by cyrilchapon** [comment]
Temporary workaround for TS users :

```
type FixedIncrementReturnType<M extends Model> = Promise<[M[], number>

MyModel.decrement(
  { balance: amount },
  { where: { id: myInstance.id } }
) as unknown as FixedIncrementReturnType<MyModel>,
```

**Comment by WikiRik** [comment]
Feel free to open a PR for this. You should already be done if you add your workaround in there and add some typing tests. Let us know if you have any questions about that

**Comment by cyrilchapon** [comment]
Was going to, but apparently it's already fixed in here : https://github.com/sequelize/sequelize/blob/main/src/model.d.ts#L2836-L2845

Is main branch supposed to reflect v6@next or v7 ?

**Comment by WikiRik** [comment]
v7 so feel free to make a PR for the v6 branch so we can fix it in 6.21.5

**Comment by fzn0x** [comment]
Thanks for reporting the issue, since it's can be a backport I would keep this as a `type: typescript` labeled issue.

Let me see if I can spare my time for this.

**Comment by fzn0x** [comment]
Let me know if this is not resolved in `6.21.6`
